chain/src/main/scala/org/bitcoins/chain/blockchain/Blockchain.scala

```diff
@@ -26,6 +26,9 @@ case class Blockchain(headers: Vector[BlockHeaderDb])
     with BitcoinSLogger {
   val tip: BlockHeaderDb = headers.head
 
+  //this could throw if all we have is the genesis
+  lazy val secondTip: BlockHeaderDb = headers(1)
```

Comment:
This seems dangerous to have in here. It's only used one place. Does that justify introducing a dangerous val? In any case, should probably have a Scaladoc explaining what the purpose here is, and that it is/why it is dangerous